AndroidHomeApplication/app/src/main/java/com/example/androidhomeapplication/data/repository/MoviesRepository.kt

```diff
@@ -0,0 +1,112 @@
+package com.example.androidhomeapplication.data.repository
+
+import com.example.androidhomeapplication.data.models.Genre
+import com.example.androidhomeapplication.data.models.Movie
+import com.example.androidhomeapplication.data.models.MovieDetails
+import com.example.androidhomeapplication.data.remote.RetrofitBuilder
+import com.example.androidhomeapplication.data.remote.response.*
+import com.example.androidhomeapplication.data.remote.services.ConfigurationService
+import com.example.androidhomeapplication.data.remote.services.MoviesService
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import retrofit2.Retrofit
+
+private const val SECURE_BASE_URL: String = "https://image.tmdb.org/t/p/"
+
+enum class ImageType(val size: String) {
+    POSTER(size = "w500"),
+    BACKDROP(size = "w780"),
+    PROFILE(size = "w185")
+}
+
+class MoviesRepository() {
+    private val retrofit: Retrofit = RetrofitBuilder.buildRetrofit()
+    private val configurationService: ConfigurationService by lazy {
+        retrofit.create(
+            ConfigurationService::class.java
+        )
+    }
+    private val movieService: MoviesService by lazy { retrofit.create(MoviesService::class.java) }
+
+    private val mutex = Mutex()
+    private var genres: Map<Long, Genre>? = null
+    private var configInfo: ConfigurationResponse? = null
+
+    suspend fun loadMovieById(movieId: Long): MovieDetails = coroutineScope {
+        val profileImageUrl = getImageUrlByType(ImageType.PROFILE)
```

Comment:
getImageUrlByType also should be async
but u can create some extension with ImageType as param for ConfigurationResponse class